### PR TITLE
Test failed because `output.zip` didn't exist.

### PR DESCRIPTION
**Pull request by dooglio**
Added guard--only call unlink if exists. Now use `fs-extra`. Added `.nvmrc`.

### CODE REVIEW

**Comment by dooglio** [comment]
Tested by running `yarn test` under Mac OS/X. Tested ACK.

File: test/index.test.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
 
 test('index: create zip file', async (t) => {
   const zipfile = 'test/output.zip'
 
   t.pass('Reading directories "fixtures"')
 
-  fs.unlinkSync(zipfile)
+  if (fs.pathExistsSync(zipfile)) {
+    fs.unlinkSync(zipfile)
+  }
```
Comment (RyanZim) [inline test/index.test.js]: Using `pathExists` before running an operation is an anti-pattern. Just run `unlinkSync`, then catch the error, if it's `ENOENT`, just ignore the error.

File: package.json [hunk 2]
```diff
@@ -22,7 +22,8 @@
   "dependencies": {
     "buffer-crc32": "^0.2.13",
     "fs-extra": "^6.0.0",
-    "klaw": "^2.1.1"
+    "klaw": "^2.1.1",
+    "yarn": "^1.7.0"
```
Comment (RyanZim) [inline package.json]: `yarn` shouldn't be installed as a dependency.

**Comment by RyanZim** [comment]
Also, committing a lockfile to git for a library is controversial, I don't think it's a good idea since it locks your deps in development but not in production.

**Comment by jprichardson** [comment]
> Also, committing a lockfile to git for a library is controversial, I don't think it's a good idea since it locks your deps in development but not in production.

I didn't think it was? https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

File: package.json (hunk 2 above)
Comment (dooglio) [reply package.json]: See: https://github.com/ExodusMovement/zip-up/pull/4

File: test/index.test.js (hunk 1 above)
Comment (dooglio) [reply test/index.test.js]: See: https://github.com/ExodusMovement/zip-up/pull/4

**Comment by dooglio** [comment]
So...what is the consensus? Remove it or keep it?

**Comment by RyanZim** [comment]
@jprichardson I know that's what yarn says, I personally disagree with their stance on the issue. I don't care enough to waste time fighting about it if you want it committed; I still think it's lame to lock deps in development when your production users are going to get something different. I highly value keeping dev and production environments as close as possible.

**Pull request merged**